cli/src/torii/routing.rs

```diff
@@ -250,36 +257,31 @@ async fn handle_post_configuration(
 }
 
 #[iroha_futures::telemetry_future]
-async fn handle_blocks_stream(wsv: &WorldStateView, mut stream: WebSocket) -> eyre::Result<()> {
+async fn handle_blocks_stream(sumeragi: Arc<Sumeragi>, mut stream: WebSocket) -> eyre::Result<()> {
     let subscription_request: VersionedBlockSubscriberMessage = stream.recv().await?;
-    let mut from_height = subscription_request.into_v1().try_into()?;
+    let mut from_height: u64 = subscription_request.into_v1().try_into()?;
 
     stream
         .send(VersionedBlockPublisherMessage::from(
             BlockPublisherMessage::SubscriptionAccepted,
         ))
         .await?;
 
-    let mut rx = wsv.subscribe_to_new_block_notifications();
-    stream_blocks(&mut from_height, wsv, &mut stream).await?;
-
     loop {
-        rx.changed().await?;
-        stream_blocks(&mut from_height, wsv, &mut stream).await?;
+        let blocks = sumeragi.blocks_from_height(from_height.try_into().unwrap());
```

Comment:
Hmm, maybe we should update `from_height` inside loop to return only new blocks?
Currently seems like we would return blocks starting from `from_height` on each cycle.

Reply:
`from_height` is mutated by `stream_blocks`.

Reply:
oh, right, mutable reference should give me a sign.